### PR TITLE
runk: Refactor container builder

### PR DESCRIPTION
**Pull request by ManaSugi**
Refactor the container builder code (`InitContainer` and `ActivatedContainer`)
to make it easier to understand and to maintain.

The details:

1. Separate the existing `builder.rs` into an `init_builder.rs` and
`activated_container.rs` to make them easy to read and maintain.

2. Move the `create_linux_container` function from the `builder.rs` to
`container.rs` because it is shared by the both files.

3. Some validation functions such as `validate_spec` from `builder.rs`
to `utils.rs` because they will be also used by other components as
utilities in the future.

Fixes: #5033

Signed-off-by: Manabu Sugimoto <Manabu.Sugimoto@sony.com>

### CODE REVIEW

**Review by liubin — APPROVED** [review]
Thank you @ManaSugi

File: src/tools/runk/libcontainer/src/activated_builder.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        if process.is_none() && args.is_empty() {
+            return Err("process and args cannot be all empty".to_string());
+        }
+        Ok(())
+    }
+}
+
+impl ActivatedContainer {
+    /// Create ContainerLauncher that can be used to spawn a process in an existing container.
+    /// It reads the spec from status file of an existing container, and adapted it with given process file
```
Comment (cyyzero) [inline src/tools/runk/libcontainer/src/activated_builder.rs:59]: Please help me fix the verb tense error.   apated -> adapte

**Review by cyyzero — APPROVED** [review]
LGTM, thank you!

File: src/tools/runk/libcontainer/src/activated_builder.rs (hunk 1 above)
Comment (ManaSugi) [reply src/tools/runk/libcontainer/src/activated_builder.rs]: I fixed it.

**Review by lifupan — APPROVED** [review]

**Pull request merged**